File: backend-python/app/models/medical_history.py
from sqlalchemy import Column, Integer, String, Date, Text
from app.database import Base

class MedicalHistory(Base):
    __tablename__ = "medical_history"
    
    id = Column(Integer, primary_key=True, index=True)
    pet_id = Column(Integer, nullable=False, index=True)
    year = Column(Integer, nullable=False)
    condition = Column(String, nullable=False)
    treatment = Column(Text)
    visit_date = Column(Date)
    hospital = Column(String)
    veterinarian = Column(String)
    notes = Column(Text)

File: backend/src/main/java/com/petcare/model/MedicalHistory.java
package com.petcare.model;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDate;

@Entity
@Table(name = "medical_history")
@Data
public class MedicalHistory {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(name = "pet_id", nullable = false)
    private Long petId;
    
    @Column(nullable = false)
    private Integer year;
    
    @Column(nullable = false)
    private String condition;
    
    @Column(columnDefinition = "TEXT")
    private String treatment;
    
    @Column(name = "visit_date")
    private LocalDate visitDate;
    
    private String hospital;
    
    private String veterinarian;
    
    @Column(columnDefinition = "TEXT")
    private String notes;
}
